docs/infrastructure/Oracles.md

```diff
@@ -63,6 +66,96 @@ The final price point for each asset is calculated by computing the assets' trad
 
 Each asset is updated every two minutes if the new price deviates from the old one by more than 0.2%. Additionally, a heartbeat of 10 minutes is applied, which means that each price is updated at least once per day even if it moves by less than 0.2%.
 
+---
+
+## Randomness Oracles
+
+### Deployed contracts
+
+The Randomness oracle contracts on Group 0 are deployed at the following addresses: 
+
+Network    | Address
+-----------|--------------------------------------------------------------
+Mainnet    | v1v4cBXP9L7M9ryZZCx7tuXuNb9pnDLGb3JJkPBpbR1Z
+Testnet    | 217k7FMPgahEQWCfSA1BN5TaxPsFovjPagpujkyxKDvS3
+
+### How the Randomness Oracle Works
+
+DIA leverages [drand](https://drand.love/)’s distributed randomness beacon, enabling verifiable, unpredictable, and unbiased random numbers.
+
+![image2](https://github.com/user-attachments/assets/85485b42-4216-41c4-baf9-bc8ad43b8425)
+
+The drand network's nodes collaborate every 30 seconds to produce a verifiable random number, which DIA's oracle system then captures and publishes to its smart contract, making this randomness readily available for on-chain applications. 
+
+You can learn more about the full publishing process [here](https://docs.diadata.org/use-nexus-product/readme/randomness-oracle/data-source#on-chain-publishing-process).
+
+### How to Access Data  
+
+#### Accessing the oracle on-chain  
+
+To consume randomness data, you’ll need to invoke the **`getLastRound`** method on the oracle contract. It will return the round ID of the latest update.  
+
+Using this round ID, you can call **`getRandomValue`** and will receive a return value of that randomness, the round ID, and the BLS signature from the drand API.  
+
+Please note that round IDs are used round-robin and will repeat after **1000 iterations**.  
+
+Below is the **DIARandomOracle** contract implementation in Ralph:  
+
+```  
+Contract DIARandomOracle(
```

Comment:
This seems to be the implementation of the `DIARandomOracle`. For developers to use it in the smart contract and fetch the randomness, do you think example like [this](https://github.com/alephium/ralph-example/blob/master/call-oracle/contracts/randomness_fetcher.ral) is more useful?